tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -420,6 +421,13 @@ decompress_chunk_exec_impl(DecompressChunkState *chunk_state, const BatchQueueFu
 			break;
 		}
 
+		if (chunk_state->has_row_marks)
+		{
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("locking compressed tuples is not supported")));
+		}
+
```

Comment:
If you want it to not error out if the compressed tuples are filtered, probably better to put it to line 439 below, above the PG projection call. I think in this place you'll miss the case when the compressed batches are fully filtered out by vectorized filters, might be good to add a test case for this.